framework/spree/utils/create-customized-fetch-fetcher.ts

```diff
@@ -0,0 +1,107 @@
+import * as qs from 'qs'
+import { errors } from '@spree/storefront-api-v2-sdk'
+import type { CreateCustomizedFetchFetcher } from '@spree/storefront-api-v2-sdk/types/interfaces/CreateCustomizedFetchFetcher'
+
+export const fetchResponseKey = Symbol('fetch-response-key')
+
+const createCustomizedFetchFetcher: CreateCustomizedFetchFetcher = (
+  fetcherOptions
+) => {
+  const { FetchError } = errors
+  const sharedHeaders = {
+    'Content-Type': 'application/json',
+  }
+
+  const { host, fetch, requestConstructor } = fetcherOptions
+
+  return {
+    fetch: async (fetchOptions) => {
+      // This fetcher always returns request equal null,
+      // because @vercel/fetch doesn't accept a Request object as argument
+      // and it's not used by NJC anyway.
+      try {
+        const { url, params, method, headers, responseParsing } = fetchOptions
+        const absoluteUrl = new URL(url, host)
+        let payload
+
+        switch (method.toUpperCase()) {
+          case 'PUT':
+          case 'POST':
+          case 'DELETE':
+          case 'PATCH':
+            payload = { body: JSON.stringify(params) }
+            break
+          default:
+            payload = null
+            absoluteUrl.search = qs.stringify(params, {
```

Comment:
Its possible to replace this with https://developer.mozilla.org/es/docs/Web/API/URLSearchParams to avoid adding a new dep?

Reply:
@goncy 👋  We removed the `qs` dependency.